pkg/irreparabledb/irreparabledb.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package irreparabledb
+
+import (
+	"context"
+
+	"storj.io/storj/internal/migrate"
+	dbx "storj.io/storj/pkg/irreparabledb/dbx"
+)
+
+// Database implements the irreparable RPC service
+type Database struct {
+	db *dbx.DB
+}
+
+// RemoteSegmentInfo is info about a single entry stored in the irreparable db
+type RemoteSegmentInfo struct {
+	SegmentPath        []byte
+	SegmentDetail      []byte
+	LostPiecesCount    int64
+	RepairUnixSec      int64
+	RepairAttemptCount int64
+}
+
+// New creates instance of Server
+func New(source string) (*Database, error) {
+	db, err := dbx.Open("sqlite3", source)
+	if err != nil {
+		return nil, err
+	}
+
+	err = migrate.Create("irreparabledb", db)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Database{
+		db: db,
+	}, nil
+}
+
+// IncrementRepairAttempts a db entry for to increment the repair attempts field
+func (db *Database) IncrementRepairAttempts(ctx context.Context, segmentInfo *RemoteSegmentInfo) (err error) {
+	_, err = db.db.Exec(`INSERT INTO irreparabledbs ( segmentpath, segmentdetail, pieces_lost_count, seg_damaged_unix_sec, repair_attempt_count) VALUES ( ?, ?, ?, ?, ? ) 
```

Comment:
sqlite and postgres have different syntax for prepared statements... `?` only works in sqlite, `$1` is needed for postgres. Take a look at `Rebind` function on dbx.

Reply:
done